RecoMET/METProducers/src/CaloRecHitsBeamHaloCleaned.cc

```diff
@@ -0,0 +1,349 @@
+   for(unsigned int i = 0;  i < eerhitsuncleaned->size(); i++){
+     const EcalRecHit & rhit = (*eerhitsuncleaned)[i];
+     bool isclean(true);
+     edm::RefVector<EcalRecHitCollection> refbeamhalorechits =  TheSummaryHalo.GetEERechits();
+     for(unsigned int j = 0; j <refbeamhalorechits.size() ; j++){
+       const EcalRecHit &rhitbeamhalo = *(refbeamhalorechits)[j];
+       if( rhit.detid() == rhitbeamhalo.detid() ) { 
+	 isclean  = false;
+	 break;
+       }
+     }
+     if(isclean) eerhitscleaned->push_back(rhit);
+   }
+
+   //  HBHERecHit
+   auto_ptr<HBHERecHitCollection> hbherhitscleaned(new HBHERecHitCollection()); 
+   for(unsigned int i = 0;  i < hbherhitsuncleaned->size(); i++){
+     const HBHERecHit & rhit = (*hbherhitsuncleaned)[i];
+     bool isclean(true);
+     edm::RefVector<HBHERecHitCollection> refbeamhalorechits =  TheSummaryHalo.GetHBHERechits();
+     for(unsigned int j = 0; j <refbeamhalorechits.size() ; j++){
+       const HBHERecHit &rhitbeamhalo = *(refbeamhalorechits)[j];
+       if( rhit.detid() == rhitbeamhalo.detid() ) { 
+	 isclean  = false;
+	 break;
+       }
+     }  
+     if(isclean) hbherhitscleaned->push_back(rhit);
+   }
+
+
+
+   iEvent.put(ebrhitscleaned,"EcalRecHitsEB");
+   iEvent.put(eerhitscleaned,"EcalRecHitsEE");
+   iEvent.put(hbherhitscleaned);
+
+}
+
+// ------------ method called once each stream before processing any runs, lumis or events  ------------
+void
+CaloRecHitsBeamHaloCleaned::beginStream(edm::StreamID)
+{
+}
+
+// ------------ method called once each stream after processing all runs, lumis and events  ------------
+void
+CaloRecHitsBeamHaloCleaned::endStream() {
+}
+
+// ------------ method called when starting to processes a run  ------------
+/*
+void
+CaloRecHitsBeamHaloCleaned::beginRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+ 
+// ------------ method called when ending the processing of a run  ------------
+/*
+void
+CaloRecHitsBeamHaloCleaned::endRun(edm::Run const&, edm::EventSetup const&)
+{
+}
+*/
+ 
+// ------------ method called when starting to processes a luminosity block  ------------
+/*
+void
+CaloRecHitsBeamHaloCleaned::beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+ 
+// ------------ method called when ending the processing of a luminosity block  ------------
+/*
+void
+CaloRecHitsBeamHaloCleaned::endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&)
+{
+}
+*/
+ 
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void
+CaloRecHitsBeamHaloCleaned::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
+  descriptions.addDefault(desc);
```

Comment:
it would be much better to actually fill this method with the config parameters and even set their defaults.